tlspuffin/src/tls/vulnerabilities.rs

```diff
@@ -39,19 +40,23 @@ pub fn seed_cve_2022_25638(server: AgentName) -> Trace<TlsQueryMatcher> {
         )
     };
 
-    // ApplicationData 0 is EncryptedExtensions
-    let certificate_request_message = term! {
-        fn_decrypt_handshake(
-            ((server, 1)[Some(TlsQueryMatcher::ApplicationData)]), // Ticket from last session
+    let decrypted_handshake = term! {
+        fn_decrypt_handshake_flight(
+            ((server, 0)/MessageFlight), // The first flight of messages sent by the server
             (fn_server_hello_transcript(((server, 0)))),
-            (fn_get_server_key_share(((server, 0)))),
+            (fn_get_server_key_share(((server, 0)[Some(TlsQueryMatcher::Handshake(Some(HandshakeType::ServerHello)))]))),
             fn_no_psk,
             fn_named_group_secp384r1,
             fn_true,
-            fn_seq_1
+            fn_seq_0  // sequence 0
         )
     };
 
+    // ApplicationData 0 is EncryptedExtensions
+    let certificate_request_message = term! {
+        fn_find_server_certificate_request((@decrypted_handshake))
```

Comment:
Note for later: we will eventually replace this kind of function symbol by queries over terms,